dev/set_matrix.py

```diff
@@ -576,6 +596,7 @@ def expand_config(config: Dict[str, Any], *, is_ref: bool = False) -> Set[Matrix
                 else:
                     install = install_dev
                 python = get_python_version(cfg.python, package_info.pip_release, DEV_VERSION)
+                runs_on = get_runs_on(cfg.runs_on, DEV_VERSION)
```

Comment:
nit: Can we just set "ubuntu-latest" here or use different version specifier? It's a bit confusing as ubuntu doesn't have "dev" version concept afaik.

Reply:
Can elaborate what's confusing here? `get_runs_on` returns "ubuntu-latest" if `cfg.runs_on` have a key matching `"dev"`.

Reply:
The confusing part is there is no chance "dev" matches with actual version.

Reply:
Actually, there is a chance. For example:

```
    runs_on:
      # For some reason, dev version doesn't work on ubuntu-24.04
      "== dev": "ubuntu-22.04"
```